### PR TITLE
Reminders example's code improvements

### PR DESCRIPTION
**Pull request by PrimeSwiftArt**

### CODE REVIEW

File: examples/reminders/lib/feature_navigation/views/navigation_page.dart [hunk 1]
```diff
@@ -16,25 +20,49 @@ class NavigationPage extends StatelessWidget implements AutoRouteWrapper {
...
-              icon: Icon(Icons.dashboard),
-            ),
-            BottomNavigationBarItem(
-              label: 'Reminders',
-              icon: Icon(Icons.list),
+  Widget build(BuildContext context) => RxBlocListener<NavigationBlocType, int>(
+        state: (bloc) => bloc.states.tabIndex,
+        listener: (context, index) {
+          switch (index) {
+            case 0:
```
Comment (StanevPrime) [inline examples/reminders/lib/feature_navigation/views/navigation_page.dart:27]: Could you please use enumeration here ?

File: examples/reminders/lib/feature_navigation/blocs/navigation_bloc.dart [hunk 2]
```diff
@@ -5,42 +5,35 @@ part 'navigation_bloc.rxb.g.dart';
...
-      .startWith(null)
-      .throttleTime(const Duration(milliseconds: 200))
-      .switchMap((value) async* {
-        ///TODO: Replace the code below with a repository invocation
-        yield Result<String>.loading();
-        await Future.delayed(const Duration(seconds: 1));
-        yield Result<String>.success('Some specific async state');
+  Stream<int> _mapToTabIndexState() => _$openTabEvent
+      .map<Result<int>>((newIndex) {
+        /// TODO: Add a real navigation permission check
```
Comment (StanevPrime) [inline examples/reminders/lib/feature_navigation/blocs/navigation_bloc.dart:25]: Could you please elaborate on the reason of this behaviour ?

File: examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:firebase_auth/firebase_auth.dart';
```
Comment (StanevPrime) [inline examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart:2]: @StanislavPrime, could you please review the changes in this file ?

File: examples/reminders/lib/base/common_blocs/coordinator_bloc.dart [hunk 4]
```diff
@@ -103,44 +101,37 @@ extension CoordinatingTasksX on CoordinatorBlocType {
...
                 )
-                .map((managedList) =>
+                .map(
+                  (managedList) => ManagedListCounterOperation<ReminderModel>(
+                      managedList: managedList,
+                      counterOperation: CounterOperation.delete),
+                ),
+            states.onReminderUpdated.whereSuccess().switchMap(
+              (event) {
+                var stream =
```
Comment (StanevPrime) [inline examples/reminders/lib/base/common_blocs/coordinator_bloc.dart:124]: This smells kinda fishy ... can't we `withLatestFromIdentifiableList` in the outer stream, something like

states.onReminderUpdated.whereSuccess().withLatestFromIdentifiableList....

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart (hunk 3 above)
Comment (StanislavPrime) [reply examples/reminders/lib/base/data_sources/remote/reminders_firebase_data_source.dart:2]: @StanevPrime I think that everything is working as expected. I tested to login with a new user, for which there were no previous collections in firebase and saw that the counters and reminders length collections are not created any more and the methods for them from the data source were deleted. So these collections are not needed any more and they can be deleted from each of the 3 firebase projects.

**Review by StanevPrime — APPROVED** [review]

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**